src/components/sbb-calendar/sbb-calendar.tsx

```diff
@@ -822,209 +827,207 @@ export class SbbCalendar implements ComponentInterface {
-
-                const isCurrentMonth =
-                  year === this._dateAdapter.getYear(this._now()) &&
-                  this._dateAdapter.getMonth(this._now()) === month.monthValue;
-
-                return (
-                  <td
-                    class={{
+          ${months.map(
+            (row: Month[]) => html`
+              <tr>
+                ${row.map((month: Month) => {
+                  const isOutOfRange = !this._isMonthInRange(month.monthValue);
+                  const isFilteredOut = !this._isMonthFilteredOut(month.monthValue);
+                  const selectedMonth = this._selected
+                    ? this._dateAdapter.getMonth(new Date(this._selected))
+                    : undefined;
+                  const selectedYear = this._selected
+                    ? this._dateAdapter.getYear(new Date(this._selected))
+                    : undefined;
+                  const selected: boolean =
+                    !!this._selected && year === selectedYear && month.monthValue === selectedMonth;
+
+                  const isCurrentMonth =
+                    year === this._dateAdapter.getYear(this._now()) &&
+                    this._dateAdapter.getMonth(this._now()) === month.monthValue;
+
+                  return html` <td
+                    class=${classMap({
                       'sbb-calendar__table-data': true,
                       'sbb-calendar__table-month': true,
-                    }}
+                    })}
                   >
                     <button
-                      class={{
+                      class=${classMap({
                         'sbb-calendar__cell': true,
                         'sbb-calendar__pill': true,
                         'sbb-calendar__cell-current': isCurrentMonth,
                         'sbb-calendar__crossed-out': !isOutOfRange && isFilteredOut,
                         'sbb-calendar__selected': selected,
-                      }}
-                      onClick={() => this._onMonthSelection(month.monthValue, year, shiftRight)}
-                      disabled={isOutOfRange || isFilteredOut}
-                      aria-label={`${month.longValue} ${this._chosenYear}`}
-                      aria-pressed={String(selected)}
-                      aria-disabled={String(isOutOfRange || isFilteredOut)}
+                      })}
+                      @click=${() => this._onMonthSelection(month.monthValue, year, shiftRight)}
+                      ?disabled=${isOutOfRange || isFilteredOut}
+                      aria-label=${`${month.longValue} ${this._chosenYear}`}
+                      aria-pressed=${selected}
+                      aria-disabled=${String(isOutOfRange || isFilteredOut)}
                       tabindex="-1"
-                      data-month={month.monthValue}
-                      onKeyDown={(evt: KeyboardEvent) => this._handleKeyboardEvent(evt)}
+                      data-month=${month.monthValue}
```

Comment:
Again, not 100% sure
```suggestion
                      data-month=${month.monthValue || nothing}
```